prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -233,16 +221,20 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
     super.nextRouteAfterSubmit(nextRoutePath);
   }
 
-  private removeIncompleteCareSettings() {
-    this.careSettings.controls
-      .forEach((control: FormGroup, index: number) => {
-        const value = control.get('careSettingCode').value;
+  private removeIncompleteCareSettings(allowEmptyCareSettings: boolean = false) {
+    this.enrolmentFormStateService.json.careSettings.forEach((careSetting: CareSetting) => {
+      const value = careSetting;
+      const index = this.careSettings.value.findIndex((value) => !value.careSettingCode);
```

Comment:
Using `value` in the outer scope and inner scope is called variable shadowing.  It should be avoided.

Reply:
fixed,